src/sql/platform/accounts/common/accountStore.ts

```diff
@@ -64,13 +64,28 @@ export default class AccountStore implements IAccountStore {
 				// Remove old accounts that are now deprecated
 				try {
 					accounts = accounts.filter(account => {
+						const accountKey = account?.key?.accountId;
 						const providerKey = account?.key?.providerId;
-						// Account has no provider, remove it.
-						if (providerKey === undefined) {
+
+						// eslint-disable-next-line eqeqeq
+						if (accountKey == undefined) {
 							return false;
 						}
+
+						// eslint-disable-next-line eqeqeq
+						if (providerKey == undefined) {
+							return false;
+						}
+
 						// Returns true if the account isn't from a deprecated provider
```

Comment:
Update this comment now since you do some other logic (probably just remove altogether - I think it's clear enough as is)